



<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
        <meta name="description" content="This contains all material for the Introduction to NGS workshop.">
      
      
      
        <meta name="author" content="Sonika Tyagi">
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.0.4, mkdocs-material-4.4.2">
    
    
      
        <title>De Novo Canu - Introduction to NGS Workshop</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/application.30686662.css">
      
      
    
    
      <script src="../../../assets/javascripts/modernizr.74668098.js"></script>
    
    
      
        <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono&display=fallback">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
    
    <link rel="stylesheet" href="../../../assets/fonts/material-icons.css">
    
    
      <link rel="stylesheet" href="../../../css/details.css">
    
    
      
    
    
  </head>
  
    <body dir="ltr">
  
    <svg class="md-svg">
      <defs>
        
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="#pacbio-reads-assembly-with-command-line-tools" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="../../.." title="Introduction to NGS Workshop" class="md-header-nav__button md-logo">
          
            <i class="md-icon"></i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            <span class="md-header-nav__topic">
              Introduction to NGS Workshop
            </span>
            <span class="md-header-nav__topic">
              
                De Novo Canu
              
            </span>
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
          
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
      <main class="md-main" role="main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href="../../.." title="Introduction to NGS Workshop" class="md-nav__button md-logo">
      
        <i class="md-icon"></i>
      
    </a>
    Introduction to NGS Workshop
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../../.." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../trainers/trainers/" title="The Trainers" class="md-nav__link">
      The Trainers
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../preamble/preamble/" title="Workshop Information" class="md-nav__link">
      Workshop Information
    </a>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4" checked>
    
    <label class="md-nav__link" for="nav-4">
      Hands-on Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        Hands-on Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../../login/login/" title="Loging to the training VM" class="md-nav__link">
      Loging to the training VM
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../R/Rhandout/" title="Introduction to R" class="md-nav__link">
      Introduction to R
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../cli/handout/handout/" title="Introduction to Command Line" class="md-nav__link">
      Introduction to Command Line
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../qc/handout/handout/" title="Quality Control" class="md-nav__link">
      Quality Control
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../alignment/handout/handout/" title="Read Alignment" class="md-nav__link">
      Read Alignment
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../rna-seq/handout/handout/" title="RNA-Seq" class="md-nav__link">
      RNA-Seq
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../denovo/handout/handout/" title="De Novo Genome Assembly" class="md-nav__link">
      De Novo Genome Assembly
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        De Novo Canu
      </label>
    
    <a href="./" title="De Novo Canu" class="md-nav__link md-nav__link--active">
      De Novo Canu
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#author-information" class="md-nav__link">
    Author Information
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources" class="md-nav__link">
    Resources
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#learning-objectives" class="md-nav__link">
    Learning objectives
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#overview" class="md-nav__link">
    Overview
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#get-data" class="md-nav__link">
    Get data
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#pacbio-and-illumina-files" class="md-nav__link">
    PacBio and Illumina files
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sample-information" class="md-nav__link">
    Sample information
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assemble" class="md-nav__link">
    Assemble
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#check-the-output" class="md-nav__link">
    Check the output
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#change-canu-parameters-if-required" class="md-nav__link">
    Change Canu parameters if required
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions" class="md-nav__link">
    Questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#trim-and-circularise" class="md-nav__link">
    Trim and circularise
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#run-circlator" class="md-nav__link">
    Run Circlator
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#check-the-output_1" class="md-nav__link">
    Check the output
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#options" class="md-nav__link">
    Options
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions_1" class="md-nav__link">
    Questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#find-smaller-plasmids" class="md-nav__link">
    Find smaller plasmids
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#align-illumina-reads-to-the-pacbio-contig" class="md-nav__link">
    Align Illumina reads to the PacBio contig
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#extract-unmapped-illumina-reads" class="md-nav__link">
    Extract unmapped Illumina reads
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#assemble-the-unmapped-reads" class="md-nav__link">
    Assemble the unmapped reads
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#trim-the-plasmid" class="md-nav__link">
    Trim the plasmid
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#plasmid-contig-orientation" class="md-nav__link">
    Plasmid contig orientation
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#collect-contigs" class="md-nav__link">
    Collect contigs
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions_2" class="md-nav__link">
    Questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#correct" class="md-nav__link">
    Correct
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#make-an-alignment-file" class="md-nav__link">
    Make an alignment file
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#run-pilon" class="md-nav__link">
    Run Pilon
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#genome-output" class="md-nav__link">
    Genome output
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions_3" class="md-nav__link">
    Questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#short-read-assembly-a-comparison" class="md-nav__link">
    Short-read assembly: a comparison
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#get-data_1" class="md-nav__link">
    Get data
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#assemble_1" class="md-nav__link">
    Assemble
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#results" class="md-nav__link">
    Results
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions_4" class="md-nav__link">
    Questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#next" class="md-nav__link">
    Next
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#further-analyses" class="md-nav__link">
    Further analyses
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#links" class="md-nav__link">
    Links
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
    
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../license/" title="License" class="md-nav__link">
      License
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#author-information" class="md-nav__link">
    Author Information
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources" class="md-nav__link">
    Resources
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#learning-objectives" class="md-nav__link">
    Learning objectives
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#overview" class="md-nav__link">
    Overview
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#get-data" class="md-nav__link">
    Get data
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#pacbio-and-illumina-files" class="md-nav__link">
    PacBio and Illumina files
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sample-information" class="md-nav__link">
    Sample information
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assemble" class="md-nav__link">
    Assemble
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#check-the-output" class="md-nav__link">
    Check the output
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#change-canu-parameters-if-required" class="md-nav__link">
    Change Canu parameters if required
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions" class="md-nav__link">
    Questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#trim-and-circularise" class="md-nav__link">
    Trim and circularise
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#run-circlator" class="md-nav__link">
    Run Circlator
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#check-the-output_1" class="md-nav__link">
    Check the output
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#options" class="md-nav__link">
    Options
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions_1" class="md-nav__link">
    Questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#find-smaller-plasmids" class="md-nav__link">
    Find smaller plasmids
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#align-illumina-reads-to-the-pacbio-contig" class="md-nav__link">
    Align Illumina reads to the PacBio contig
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#extract-unmapped-illumina-reads" class="md-nav__link">
    Extract unmapped Illumina reads
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#assemble-the-unmapped-reads" class="md-nav__link">
    Assemble the unmapped reads
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#trim-the-plasmid" class="md-nav__link">
    Trim the plasmid
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#plasmid-contig-orientation" class="md-nav__link">
    Plasmid contig orientation
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#collect-contigs" class="md-nav__link">
    Collect contigs
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions_2" class="md-nav__link">
    Questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#correct" class="md-nav__link">
    Correct
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#make-an-alignment-file" class="md-nav__link">
    Make an alignment file
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#run-pilon" class="md-nav__link">
    Run Pilon
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#genome-output" class="md-nav__link">
    Genome output
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions_3" class="md-nav__link">
    Questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#short-read-assembly-a-comparison" class="md-nav__link">
    Short-read assembly: a comparison
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#get-data_1" class="md-nav__link">
    Get data
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#assemble_1" class="md-nav__link">
    Assemble
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#results" class="md-nav__link">
    Results
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions_4" class="md-nav__link">
    Questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#next" class="md-nav__link">
    Next
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#further-analyses" class="md-nav__link">
    Further analyses
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#links" class="md-nav__link">
    Links
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                <h1 id="pacbio-reads-assembly-with-command-line-tools">Pacbio reads: assembly with command line tools</h1>
<p><em>Keywords: de novo assembly, PacBio, PacificBiosciences, Illumina, command line, Canu, Circlator, BWA, Spades, Pilon, Microbial Genomics Virtual Laboratory</em></p>
<p>This tutorial demonstrates how to use long Pacbio sequence reads to assemble a bacterial genome, including correcting the assembly with short Illumina reads.</p>
<h2 id="author-information">Author Information</h2>
<p><em>Primary Author(s):</em>
Anna Syme Melbourne Bioinformatics</p>
<h2 id="resources">Resources</h2>
<p>Tools (and versions) used in this tutorial include:</p>
<ul>
<li>canu 1.5 [recently updated]</li>
<li>infoseq and sizeseq (part of EMBOSS) 6.6.0.0</li>
<li>circlator 1.5.1 [recently updated]</li>
<li>bwa 0.7.15</li>
<li>samtools 1.3.1</li>
<li>makeblastdb and blastn (part of blast) 2.4.0+</li>
<li>pilon 1.20</li>
</ul>
<h2 id="learning-objectives">Learning objectives</h2>
<p>At the end of this tutorial, be able to:</p>
<ol>
<li>Assemble and circularise a bacterial genome from PacBio sequence data.</li>
<li>Recover small plasmids missed by long read sequencing, using Illumina data</li>
<li>Explore the effect of polishing assembled sequences with a different data set.</li>
</ol>
<h2 id="overview">Overview</h2>
<p>Simplified version of workflow:</p>
<p><img alt="workflow" src="../images/flowchart.png" /></p>
<h2 id="get-data">Get data</h2>
<p>The files we need are here [ In a new tab, go to <a href="https://doi.org/10.5281/zenodo.1009308">https://doi.org/10.5281/zenodo.1009308</a>]:</p>
<ul>
<li><fn>pacbio.fq</fn> : the PacBio reads</li>
<li><fn>R1.fq</fn>: the Illumina forward reads</li>
<li><fn>R2.fq</fn>: the Illumina reverse reads</li>
</ul>
<p>If you already have the files, skip forward to next section, <a href="#assemble">Assemble</a>.</p>
<p>Otherwise, this section has information about how to find and move the files:</p>
<h3 id="pacbio-and-illumina-files">PacBio and Illumina files</h3>
<p>In a new tab, go to <a href="https://doi.org/10.5281/zenodo.1009308">https://doi.org/10.5281/zenodo.1009308</a>.</p>
<p>Next to the first file, right-click (or control-click) the &ldquo;Download&rdquo; button, and select &ldquo;Copy link address&rdquo;.
Back in your terminal, enter</p>
<p><div class="codehilite"><pre><span></span>wget  [paste link URL for file]
</pre></div>
The file should download.
Note: paste the link to the file, not to the webpage.
Repeat this for the other two files.
Shorten each of these files names with the mv command:</p>
<div class="codehilite"><pre><span></span>mv R1.fq\?download\=1 R1.fq
mv R2.fq\?download\=1 R2.fq
mv pacbio.fq\?download\=1 pacbio.fq
</pre></div>

<p>Type in <code class="codehilite"><span class="n">ls</span></code> to check the files are present and correctly-named.</p>
<p>We should have <code class="codehilite"><span class="n">R1</span><span class="p">.</span><span class="n">fq</span></code>, <code class="codehilite"><span class="n">R2</span><span class="p">.</span><span class="n">fq</span></code> and <code class="codehilite"><span class="n">pacbio</span><span class="p">.</span><span class="n">fq</span></code>.</p>
<div class="codehilite"><pre><span></span>cd denovo-canu
ls
</pre></div>

<h3 id="sample-information">Sample information</h3>
<p>The sample used in this tutorial is a gram-positive bacteria called <em>Staphylococcus aureus</em> (sample number 25747). This particular sample is from a strain that is resistant to the antibiotic methicillin (a type of penicillin). It is also called MRSA: methicillin-resistant <em>Staphylococcus aureus</em>. It was isolated from (human) blood and caused bacteraemia, an infection of the bloodstream.</p>
<h2 id="assemble">Assemble<a name="assemble"></a></h2>
<ul>
<li>We will use the assembly software called <a href="http://canu.readthedocs.io/en/stable/">Canu</a>.</li>
<li>Run Canu with these commands:</li>
</ul>
<div class="codehilite"><pre><span></span>canu -p canu -d canu_outdir genomeSize=0.03m -pacbio-raw pacbio.fq
</pre></div>

<ul>
<li>the first <code class="codehilite"><span class="n">canu</span></code> tells the program to run</li>
<li><code class="codehilite"><span class="o">-</span><span class="n">p</span> <span class="n">canu</span></code> names prefix for output files (&ldquo;canu&rdquo;)</li>
<li><code class="codehilite"><span class="o">-</span><span class="n">d</span> <span class="n">canu_outdir</span></code> names output directory (&ldquo;canu_outdir&rdquo;)</li>
<li>
<p><code class="codehilite"><span class="n">genomeSize</span></code> only has to be approximate.</p>
<ul>
<li>e.g. <em>Staphylococcus aureus</em>, 2.8m</li>
<li>e.g. <em>Streptococcus pyogenes</em>, 1.8m</li>
<li>(In this case we are using a partial genome of expected size 30,000 base pairs).</li>
</ul>
</li>
<li>
<p>Canu will correct, trim and assemble the reads.</p>
</li>
<li>Various output will be displayed on the screen.</li>
</ul>
<h3 id="check-the-output">Check the output</h3>
<p>Move into <fn>canu_outdir</fn> and <code class="codehilite"><span class="n">ls</span></code> to see the output files.</p>
<ul>
<li>The <fn>canu.contigs.fasta</fn> are the assembled sequences.</li>
<li>The <fn>canu.unassembled.fasta</fn> are the reads that could not be assembled.</li>
<li>The <fn>canu.correctedReads.fasta.gz</fn> are the corrected Pacbio reads that were used in the assembly.</li>
<li>The <fn>canu.file.gfa</fn> is the graph of the assembly.</li>
<li>Display summary information about the contigs: (<code class="codehilite"><span class="n">infoseq</span></code> is a tool from <a href="http://emboss.sourceforge.net/index.html">EMBOSS</a>)</li>
</ul>
<div class="codehilite"><pre><span></span>infoseq canu.contigs.fasta
</pre></div>

<ul>
<li>This will show the contigs found by Canu. e.g.,</li>
</ul>
<div class="codehilite"><pre><span></span>    - tig00000001   47997
</pre></div>

<ul>
<li>This will show the contigs found by Canu. e.g., tig00000001 47997
    -&ldquo;tig00000001&rdquo; is the name given to the contig
    -&ldquo;47997&rdquo; is the number of base pairs in that contig.</li>
</ul>
<p>This matches what we were expecting for this sample (approximately 30,000 base pairs). For other data, Canu may not be able to join all the reads into one contig, so there may be several contigs in the output.</p>
<p>We should also look at the canu.report. To do this:</p>
<p><div class="codehilite"><pre><span></span>less canu.report
</pre></div>
- <code class="codehilite"><span class="k">less</span></code> is a command to display the file on the screen.
- Use the up and down arrows to scroll up and down.
- You will see lots of histograms of read lengths before and after processing, final contig construction, etc.
- For a description of the outputs that Canu produces, see: <a href="http://canu.readthedocs.io/en/latest/tutorial.html#outputs">http://canu.readthedocs.io/en/latest/tutorial.html#outputs</a>
- Type <code class="codehilite"><span class="n">q</span></code> to exit viewing the report</p>
<h3 id="change-canu-parameters-if-required">Change Canu parameters if required</h3>
<p>If the assembly is poor with many contigs, re-run Canu with extra sensitivity parameters; e.g.
<div class="codehilite"><pre><span></span>canu -p prefix -d outdir corMhapSensitivity=high corMinCoverage=0 genomeSize=.03m -pacbio-raw pacbio.fq
</pre></div></p>
<h3 id="questions">Questions</h3>
<p>Q: How do long- and short-read assembly methods differ? A: short reads: De Bruijn graphs; long reads: a move back towards simpler overlap-layout-consensus methods.</p>
<p>Q: Where can we find out the what the approximate genome size should be for the species being assembled? A: NCBI Genomes - enter species name - click on Genome Assembly and Annotation report - sort table by clicking on the column header Size (Mb) - look at range of sizes in this column.</p>
<p>Q: In the assembly output, what are the unassembled reads? Why are they there?</p>
<p>Q: What are the corrected reads? How did canu correct the reads?</p>
<p>Q: Where could you view the output .gfa and what would it show?</p>
<h2 id="trim-and-circularise">Trim and circularise</h2>
<h3 id="run-circlator">Run Circlator</h3>
<p>Circlator identifies and trims overhangs (on chromosomes and plasmids) and orients the start position at an appropriate gene (e.g. dnaA). It takes in the assembled contigs from Canu, as well as the corrected reads prepared by Canu.</p>
<p>Overhangs are shown in blue:</p>
<p><img alt="circlator" src="../images/circlator_diagram.png" />
<em>Adapted from Figure 1. Hunt et al. Genome Biology 2015</em></p>
<p>Move back into your main analysis folder.</p>
<p>Run Circlator:</p>
<div class="codehilite"><pre><span></span>circlator all --threads 8 --verbose canu_outdir/canu.contigs.fasta canu_outdir/canu.correctedReads.fasta.gz circlator_outdir
</pre></div>

<ul>
<li><code class="codehilite"><span class="c1">--threads</span></code> is the number of cores: change this to an appropriate number</li>
<li><code class="codehilite"><span class="c1">--verbose</span></code> prints progress information to the screen</li>
<li><code class="codehilite"><span class="n">canu_outdir</span><span class="o">/</span><span class="n">canu</span><span class="p">.</span><span class="n">contigs</span><span class="p">.</span><span class="n">fasta</span></code> is the file path to the input Canu assembly</li>
<li><code class="codehilite"><span class="n">canu_outdir</span><span class="o">/</span><span class="n">canu</span><span class="p">.</span><span class="n">correctedReads</span><span class="p">.</span><span class="n">fasta</span><span class="p">.</span><span class="n">gz</span></code> is the file path to the corrected Pacbio reads - note, fastA not fastQ</li>
<li><code class="codehilite"><span class="n">circlator_outdir</span></code> is the name of the output directory.</li>
</ul>
<p>Some output will print to screen. When finished, it should say &ldquo;Circularized x of x contig(s)&rdquo;.</p>
<h3 id="check-the-output_1">Check the output</h3>
<p>Move into the <fn>circlator_outdir</fn> directory and <code class="codehilite"><span class="n">ls</span></code> to list files.</p>
<p><em>Were the contigs circularised?</em> :</p>
<div class="codehilite"><pre><span></span>less 04.merge.circularise.log
</pre></div>

<ul>
<li>Yes, the contig was circularised (last column).</li>
<li>Type &ldquo;q&rdquo; to exit.</li>
</ul>
<p><em>Where were the contigs oriented (which gene)?</em> :</p>
<p><div class="codehilite"><pre><span></span>less 06.fixstart.log
</pre></div>
- Look in the &ldquo;gene_name&rdquo; column.
- The contig has been oriented at tr|A0A090N2A8|A0A090N2A8_STAAU, which is another name for dnaA. <!-- (search swissprot - uniprot.org) --> This is typically used as the start of bacterial chromosome sequences.</p>
<p><em>What are the trimmed contig sizes?</em> :</p>
<div class="codehilite"><pre><span></span>infoseq 06.fixstart.fasta
</pre></div>

<ul>
<li>tig00000001 30019 (bases trimmed)</li>
</ul>
<p>This trimmed part is the overlap.</p>
<p><em>Re-name the contigs file</em>:</p>
<ul>
<li>The trimmed contigs are in the file called <fn>06.fixstart.fasta</fn>.</li>
<li>Re-name it <fn>contig1.fasta</fn>:</li>
</ul>
<div class="codehilite"><pre><span></span>mv 06.fixstart.fasta contig1.fasta
</pre></div>

<p>Open this file in a text editor (e.g. nano: <code class="codehilite"><span class="n">nano</span> <span class="n">contig1</span><span class="p">.</span><span class="n">fasta</span></code>) and change the header to &ldquo;&gt;chromosome&rdquo;.</p>
<p>Move the file back into the main folder (<code class="codehilite"><span class="n">mv</span> <span class="n">contig1</span><span class="p">.</span><span class="n">fasta</span> <span class="p">..</span><span class="o">/</span></code>).</p>
<h3 id="options">Options</h3>
<p>If all the contigs have not circularised with Circlator, an option is to change the <code class="codehilite"><span class="c1">--b2r_length_cutoff</span></code> setting to approximately 2X the average read depth.</p>
<h3 id="questions_1">Questions</h3>
<p>Q: Were all the contigs circularised? Why/why not?</p>
<p>Q: Circlator can set the start of the sequence at a particular gene. Which gene does it use? Is this appropriate for all contigs? A: Uses dnaA for the chromosomal contig. For other contigs, uses a centrally-located gene. However, ideally, plasmids would be oriented on a gene such as repA. It is possible to provide a file to Circlator to do this.</p>
<h2 id="find-smaller-plasmids">Find smaller plasmids</h2>
<p>Pacbio reads are long, and may have been longer than small plasmids. We will look for any small plasmids using the Illumina reads.</p>
<p>This section involves several steps:</p>
<ol>
<li>Use the Canu+Circlator output of a trimmed assembly contig.</li>
<li>Map all the Illumina reads against this Pacbio-assembled contig.</li>
<li>Extract any reads that <em>didn&rsquo;t</em> map and assemble them together: this could be a plasmid, or part of a plasmid.</li>
<li>Look for overhang: if found, trim.</li>
</ol>
<h3 id="align-illumina-reads-to-the-pacbio-contig">Align Illumina reads to the PacBio contig</h3>
<ul>
<li>Index the contigs file:</li>
</ul>
<div class="codehilite"><pre><span></span>bwa index contig1.fasta
</pre></div>

<ul>
<li>Align Illumina reads using using bwa mem:</li>
</ul>
<div class="codehilite"><pre><span></span>bwa mem -t 8 contig1.fasta R1.fq R2.fq | samtools sort &gt; aln.bam
</pre></div>

<ul>
<li><code class="codehilite"><span class="n">bwa</span> <span class="n">mem</span></code> is the alignment tool</li>
<li><code class="codehilite"><span class="o">-</span><span class="n">t</span> <span class="mi">8</span></code> is the number of cores: choose an appropriate number</li>
<li><code class="codehilite"><span class="n">contig1</span><span class="p">.</span><span class="n">fasta</span></code> is the input assembly file</li>
<li><code class="codehilite"><span class="n">R1</span><span class="p">.</span><span class="n">fq</span> <span class="n">R2</span><span class="p">.</span><span class="n">fq</span></code> are the Illumina reads</li>
<li><code class="codehilite"><span class="o">|</span> <span class="n">samtools</span> <span class="n">sort</span></code> pipes the output to samtools to sort</li>
<li><code class="codehilite"><span class="o">&gt;</span> <span class="n">aln</span><span class="p">.</span><span class="n">bam</span></code> sends the alignment to the file <fn>aln.bam</fn></li>
</ul>
<h3 id="extract-unmapped-illumina-reads">Extract unmapped Illumina reads</h3>
<ul>
<li>Index the alignment file:</li>
</ul>
<div class="codehilite"><pre><span></span>samtools index aln.bam
</pre></div>

<ul>
<li>Extract the fastq files from the bam alignment - those reads that were unmapped to the Pacbio alignment - and save them in various &ldquo;unmapped&rdquo; files:</li>
</ul>
<div class="codehilite"><pre><span></span>samtools fastq -f 4 -1 unmapped.R1.fastq -2 unmapped.R2.fastq -s unmapped.RS.fastq aln.bam
</pre></div>

<ul>
<li><code class="codehilite"><span class="n">fastq</span></code> is a command that coverts a <fn>.bam</fn> file into fastq format</li>
<li><code class="codehilite"><span class="o">-</span><span class="n">f</span> <span class="mi">4</span></code> : only output unmapped reads</li>
<li><code class="codehilite"><span class="o">-</span><span class="mi">1</span></code> : put R1 reads into a file called <fn>unmapped.R1.fastq</fn></li>
<li><code class="codehilite"><span class="o">-</span><span class="mi">2</span></code> : put R2 reads into a file called <fn>unmapped.R2.fastq</fn></li>
<li><code class="codehilite"><span class="o">-</span><span class="n">s</span></code> : put singleton reads into a file called <fn>unmapped.RS.fastq</fn></li>
<li><code class="codehilite"><span class="n">aln</span><span class="p">.</span><span class="n">bam</span></code> : input alignment file</li>
</ul>
<p>We now have three files of the unampped reads: <fn> unmapped.R1.fastq</fn>, <fn> unmapped.R2.fastq</fn>, <fn> unmapped.RS.fastq</fn>.</p>
<h3 id="assemble-the-unmapped-reads">Assemble the unmapped reads</h3>
<ul>
<li>Assemble with Spades:</li>
</ul>
<div class="codehilite"><pre><span></span>spades.py -1 unmapped.R1.fastq -2 unmapped.R2.fastq -s unmapped.RS.fastq --careful --cov-cutoff auto -o spades_assembly
</pre></div>

<ul>
<li><code class="codehilite"><span class="o">-</span><span class="mi">1</span></code> is input file forward</li>
<li><code class="codehilite"><span class="o">-</span><span class="mi">2</span></code> is input file reverse</li>
<li><code class="codehilite"><span class="o">-</span><span class="n">s</span></code> is unpaired</li>
<li><code class="codehilite"><span class="c1">--careful</span></code> minimizes mismatches and short indels</li>
<li><code class="codehilite"><span class="c1">--cov-cutoff auto</span></code> computes the coverage threshold (rather than the default setting, &ldquo;off&rdquo;)</li>
<li><code class="codehilite"><span class="o">-</span><span class="n">o</span></code> is the output directory</li>
</ul>
<p>Move into the output directory (<fn>spades_assembly</fn>) and look at the contigs:</p>
<p><div class="codehilite"><pre><span></span>cd spades_assembly
infoseq contigs.fasta
</pre></div>
- 1 contig has been assembled with a length of 2359 bases.
- Type &ldquo;q&rdquo; to exit.</p>
<p>Copy it to a new file:</p>
<div class="codehilite"><pre><span></span>cp contigs.fasta contig2.fasta
</pre></div>

<h3 id="trim-the-plasmid">Trim the plasmid</h3>
<p>To trim any overhang on this plasmid, we will blast the start of contig2 against itself.</p>
<ul>
<li>Take the start of the contig:</li>
</ul>
<div class="codehilite"><pre><span></span>head -n 10 contig2.fasta &gt; contig2.fa.head
</pre></div>

<ul>
<li>head -n 10 takes the first ten lines of <fn>contig2.fasta</fn></li>
<li><code class="codehilite"><span class="o">&gt;</span></code> sends that output to a new file called <fn>contig2.fa.head</fn></li>
<li>We want to see if it matches the end (overhang).</li>
<li>Format the assembly file for blast:</li>
</ul>
<div class="codehilite"><pre><span></span>makeblastdb -in contig2.fasta -dbtype nucl
</pre></div>

<ul>
<li>
<p>Blast the start of the assembly (.head file) against all of the assembly:
<div class="codehilite"><pre><span></span>blastn -query contig2.fa.head -db contig2.fasta -evalue 1e-3 -dust no -out contig2.bls
</pre></div></p>
</li>
<li>
<p><code class="codehilite"><span class="n">blastn</span></code> is the tool Blast, set as blastn to compare sequences of nucleotides to each other</p>
</li>
<li><code class="codehilite"><span class="n">query</span></code> sets the input sequence as <fn>contig2.fa.head</fn></li>
<li><code class="codehilite"><span class="n">db</span></code> sets the database as that of the original sequence <fn>contig2.fasta</fn>. We don&rsquo;t have to specify the other files that were created when we formatted this file, but they need to present in our current directory.</li>
<li><code class="codehilite"><span class="n">evalue</span></code> is the number of hits expected by chance, here set as 1e-3</li>
<li><code class="codehilite"><span class="n">dust</span> <span class="k">no</span></code> turns off the masking of low-complexity regions</li>
<li>
<p><code class="codehilite"><span class="k">out</span></code> sets the output file as <fn>contig2.bls</fn></p>
</li>
<li>
<p>Look at <fn>contig2.bls</fn> to see hits:</p>
</li>
</ul>
<div class="codehilite"><pre><span></span>less contig2.bls
</pre></div>

<ul>
<li>The first hit is at start, as expected. We can see that &ldquo;Query 1&rdquo; (the start of the contig) is aligned to &ldquo;Sbject 1&rdquo; (the whole contig), for the first 540 bases.</li>
<li>Scroll down with the down arrow.</li>
<li>The second hit shows &ldquo;Query 1&rdquo; (the start of the contig) also matches to &ldquo;Sbject 1&rdquo; (the whole contig) at position 2253, all the way to the end, position 2359. </li>
<li>This is the overhang.</li>
<li>Trim to position 2252.</li>
<li>Index the plasmid.fa file:</li>
</ul>
<p>First, change the name of the contig within the file:</p>
<p><div class="codehilite"><pre><span></span>nano contig2.fasta
</pre></div>
- <code class="codehilite"><span class="n">nano</span></code> opens up a text editor.
- Use the arrow keys to navigate. (The mouse won&rsquo;t work.)
- At the first line, delete the text, which will be something like <code class="codehilite"><span class="o">&gt;</span><span class="n">NODE_1_length_2359_cov_3</span><span class="p">.</span><span class="mi">320333</span></code>
- Type in <code class="codehilite"><span class="o">&gt;</span><span class="n">contig2</span></code>
- Don&rsquo;t forget the <code class="codehilite"><span class="o">&gt;</span></code> symbol
- Press <code class="codehilite"><span class="n">Control</span><span class="o">-</span><span class="n">X</span></code>
- <code class="codehilite"><span class="n">Save</span> <span class="n">modified</span> <span class="n">buffer</span> <span class="o">?</span></code> - type Y
- Press the <code class="codehilite"><span class="n">Enter</span></code> key</p>
<p>Index the file (this will allow samtools to edit the file as it will have an index):
<div class="codehilite"><pre><span></span>samtools faidx contig2.fasta
</pre></div></p>
<ul>
<li>Trim the contig:
<div class="codehilite"><pre><span></span>samtools faidx contig2.fasta plasmid:1-2252 &gt; plasmid.fa.trimmed
</pre></div></li>
<li>
<p><code class="codehilite"><span class="n">plasmid</span></code> is the name of the contig, and we want the sequence from 1-2252.</p>
</li>
<li>
<p>Open this file in nano (<code class="codehilite"><span class="n">nano</span> <span class="n">plasmid</span><span class="p">.</span><span class="n">fa</span><span class="p">.</span><span class="n">trimmed</span></code>) and change the header to &ldquo;&gt;plasmid&rdquo;, save.</p>
</li>
<li>We now have a trimmed plasmid.</li>
<li>Move file back into main folder:</li>
</ul>
<div class="codehilite"><pre><span></span>cp plasmid.fa.trimmed ../
</pre></div>

<ul>
<li>Move into the main folder.</li>
</ul>
<h3 id="plasmid-contig-orientation">Plasmid contig orientation</h3>
<p>The bacterial chromosome was oriented at the gene dnaA. Plasmids are often oriented at the replication gene, but this is highly variable and there is no established convention. Here we will orient the plasmid at a gene found by Prodigal, in Circlator:</p>
<div class="codehilite"><pre><span></span>circlator fixstart plasmid.fa.trimmed plasmid_fixstart
</pre></div>

<ul>
<li><code class="codehilite"><span class="n">fixstart</span></code> is an option in Circlator just to orient a sequence.</li>
<li><code class="codehilite"><span class="n">plasmid</span><span class="p">.</span><span class="n">fa</span><span class="p">.</span><span class="n">trimmed</span></code> is our small plasmid.</li>
<li><code class="codehilite"><span class="n">plasmid_fixstart</span></code> is the prefix for the output files.</li>
</ul>
<p>View the output:</p>
<div class="codehilite"><pre><span></span>less plasmid_fixstart.log
</pre></div>

<ul>
<li>The plasmid has been oriented at a gene predicted by Prodigal, and the break-point is at position 1200.</li>
<li>Change the file name:</li>
</ul>
<div class="codehilite"><pre><span></span>cp plasmid_fixstart.fasta contig2.fasta
</pre></div>

<!-- note: annotated with prokka. plasmid only has two proteins. ermC, and a hypothetical protein. protein blast genbank: matches a staph replication and maintenance protein. -->

<h3 id="collect-contigs">Collect contigs</h3>
<div class="codehilite"><pre><span></span>cat contig1.fasta contig2.fasta &gt; genome.fasta
</pre></div>

<ul>
<li>
<p>See the contigs and sizes:
<div class="codehilite"><pre><span></span>infoseq genome.fasta
</pre></div></p>
</li>
<li>
<p>chromosome: 2823331</p>
</li>
<li>plasmid: 2473</li>
</ul>
<h3 id="questions_2">Questions</h3>
<p>Q: Why is this section so complicated? A: Finding small plasmids is difficult for many reasons! This paper has a nice summary: On the (im)possibility to reconstruct plasmids from whole genome short-read sequencing data. doi: <a href="https://doi.org/10.1101/086744">https://doi.org/10.1101/086744</a></p>
<p>Q: Why can PacBio sequencing miss small plasmids? A: Library prep size selection</p>
<p>Q: We extract unmapped Illumina reads and assemble these to find small plasmids. What could they be missing? A: Repeats that have mapped to the PacBio assembly.</p>
<p>Q: How do you find a plasmid in a Bandage graph? A: It is probably circular, matches the size of a known plasmid, has a rep gene&hellip;</p>
<p>Q: Are there easier ways to find plasmids? A: Possibly. One option is the program called Unicycler which may automate many of these steps. <a href="https://github.com/rrwick/Unicycler">https://github.com/rrwick/Unicycler</a></p>
<h2 id="correct">Correct</h2>
<p>We will correct the Pacbio assembly with Illumina reads.</p>
<h3 id="make-an-alignment-file">Make an alignment file</h3>
<ul>
<li>Align the Illumina reads (R1 and R2) to the draft PacBio assembly, e.g. <fn>genome.fasta</fn>:</li>
</ul>
<div class="codehilite"><pre><span></span>bwa index genome.fasta
bwa mem -t 32 genome.fasta R1.fq R2.fq | samtools sort &gt; aln.bam
</pre></div>

<ul>
<li>
<p><code class="codehilite"><span class="o">-</span><span class="n">t</span></code> is the number of cores: set this to an appropriate number. (To find out how many you have, <code class="codehilite"><span class="n">grep</span> <span class="o">-</span><span class="k">c</span> <span class="n">processor</span> <span class="o">/</span><span class="n">proc</span><span class="o">/</span><span class="n">cpuinfo</span></code>).</p>
</li>
<li>
<p>Index the files:</p>
</li>
</ul>
<div class="codehilite"><pre><span></span>samtools index aln.bam
samtools faidx genome.fasta
</pre></div>

<ul>
<li>Now we have an alignment file to use in Pilon: <fn>aln.bam</fn></li>
</ul>
<h3 id="run-pilon">Run Pilon</h3>
<ul>
<li>Run:</li>
</ul>
<div class="codehilite"><pre><span></span>pilon --genome genome.fasta --frags aln.bam --output pilon1 --fix all --mindepth 0.5 --changes --verbose --threads 32
</pre></div>

<ul>
<li><code class="codehilite"><span class="c1">--genome</span></code> is the name of the input assembly to be corrected</li>
<li><code class="codehilite"><span class="c1">--frags</span></code> is the alignment of the reads against the assembly</li>
<li><code class="codehilite"><span class="c1">--output</span></code> is the name of the output prefix</li>
<li><code class="codehilite"><span class="c1">--fix</span></code> is an option for types of corrections</li>
<li><code class="codehilite"><span class="c1">--mindepth</span></code> gives a minimum read depth to use</li>
<li><code class="codehilite"><span class="c1">--changes</span></code> produces an output file of the changes made</li>
<li><code class="codehilite"><span class="c1">--verbose</span></code> prints information to the screen during the run</li>
<li><code class="codehilite"><span class="c1">--threads</span></code> : set this to an appropriate number</li>
</ul>
<p>Look at the changes file:</p>
<div class="codehilite"><pre><span></span>less pilon1.changes
</pre></div>

<p><em>Example:</em></p>
<p><img alt="pilon" src="../images/pilon.png" /></p>
<p>Look at the details of the fasta file:</p>
<div class="codehilite"><pre><span></span>infoseq pilon1.fasta
</pre></div>

<ul>
<li>chromosome - 2823340 (net +9 bases)</li>
<li>plasmid - 2473 (no change)</li>
</ul>
<p><strong>Option:</strong></p>
<p>If there are many changes, run Pilon again, using the <fn>pilon1.fasta</fn> file as the input assembly, and the Illumina reads to correct.</p>
<h3 id="genome-output">Genome output</h3>
<ul>
<li>Change the file name:</li>
</ul>
<div class="codehilite"><pre><span></span>cp pilon1.fasta assembly.fasta
</pre></div>

<ul>
<li>We now have the corrected genome assembly of <em>Staphylococcus aureus</em> in .fasta format, containing a chromosome and a small plasmid.  </li>
</ul>
<h3 id="questions_3">Questions</h3>
<p>Q: Why don&rsquo;t we correct earlier in the assembly process? A: We need to circularise the contigs and trim overhangs first.</p>
<p>Q: Why can we use some reads (Illumina) to correct other reads (PacBio) ? A: Illumina reads have higher accuracy</p>
<p>Q: Could we just use PacBio reads to assemble the genome? A: Yes, if accuracy adequate.</p>
<h2 id="short-read-assembly-a-comparison">Short-read assembly: a comparison</h2>
<p>So far, we have assembled the long PacBio reads into one contig (the chromosome) and found an additional plasmid in the Illumina short reads.</p>
<p>If we only had Illumina reads, we could also assemble these using the tool Spades.</p>
<p>You can try this here or try it later on your own data.</p>
<h3 id="get-data_1">Get data</h3>
<p>We will use the same Illumina data as we used above:</p>
<ul>
<li><fn>R1.fq</fn>: the Illumina forward reads</li>
<li><fn>R2.fq</fn>: the Illumina reverse reads</li>
</ul>
<h3 id="assemble_1">Assemble</h3>
<p>Run Spades:</p>
<div class="codehilite"><pre><span></span>spades.py -1 R1.fq -2 R2.fq --careful --cov-cutoff auto -o spades_assembly_all_illumina
</pre></div>

<ul>
<li><code class="codehilite"><span class="o">-</span><span class="mi">1</span></code> is input file of forward reads</li>
<li><code class="codehilite"><span class="o">-</span><span class="mi">2</span></code> is input file of reverse reads</li>
<li><code class="codehilite"><span class="c1">--careful</span></code> minimizes mismatches and short indels</li>
<li><code class="codehilite"><span class="c1">--cov-cutoff auto</span></code> computes the coverage threshold (rather than the default setting, &ldquo;off&rdquo;)</li>
<li><code class="codehilite"><span class="o">-</span><span class="n">o</span></code> is the output directory</li>
</ul>
<h3 id="results">Results</h3>
<p>Move into the output directory and look at the contigs:</p>
<div class="codehilite"><pre><span></span>infoseq contigs.fasta
</pre></div>

<h3 id="questions_4">Questions</h3>
<p>How many contigs were found by Spades?</p>
<ul>
<li>many</li>
</ul>
<p>How does this compare to the number of contigs found by assembling the long read data with Canu?</p>
<ul>
<li>many more.</li>
</ul>
<p>Does it matter that an assembly is in many contigs? </p>
<ul>
<li>
<p>Yes</p>
</li>
<li>
<p>broken genes =&gt; missing/incorrect annotations</p>
</li>
<li>
<p>less information about structure: e.g. number of plasmids</p>
</li>
<li>
<p>No</p>
</li>
<li>
<p>Many or all genes may still be annotated</p>
</li>
<li>Gene location is useful (e.g. chromosome, plasmid1) but not always essential (e.g. presence/absence of particular resistance genes)</li>
</ul>
<p>How can we get more information about the assembly from Spades?</p>
<ul>
<li>Look at the assembly graph <fn>assembly_graph.fastg</fn>, e.g. in the program Bandage. This shows how contigs are related, albeit with ambiguity in some places. </li>
</ul>
<h2 id="next">Next</h2>
<h3 id="further-analyses">Further analyses</h3>
<ul>
<li>Annotate with Prokka.</li>
<li>Comparative genomics, e.g. with Roary.</li>
</ul>
<h3 id="links">Links</h3>
<ul>
<li><a href="https://s3.amazonaws.com/files.pacb.com/software/instrument/2.0.0/bas.h5+Reference+Guide.pdf">Details of bas.h5 files</a></li>
<li>Canu <a href="http://canu.readthedocs.io/en/stable/quick-start.html">manual</a> and <a href="https://github.com/marbl/canu">gitub repository</a></li>
<li>Circlator <a href="http://genomebiology.biomedcentral.com/articles/10.1186/s13059-015-0849-0">article</a> and <a href="http://sanger-pathogens.github.io/circlator/">github repository</a></li>
<li>Pilon <a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0112963">article</a> and <a href="https://github.com/broadinstitute/pilon/wiki">github repository</a></li>
<li>Notes on <a href="https://github.com/PacificBiosciences/Bioinformatics-Training/wiki/Finishing-Bacterial-Genomes">finishing</a> and <a href="https://github.com/PacificBiosciences/Bioinformatics-Training/wiki/Evaluating-Assemblies">evaluating</a> assemblies.</li>
</ul>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../../../denovo/handout/handout/" title="De Novo Genome Assembly" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                De Novo Genome Assembly
              </span>
            </div>
          </a>
        
        
          <a href="../../../license/" title="License" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                License
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            &copy; The authors; 2019-
          </div>
        
        powered by
        <a href="https://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../../assets/javascripts/application.c648116f.js"></script>
      
      <script>app.initialize({version:"1.0.4",url:{base:"../../.."}})</script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
      
        <script src="../../../js/details.js"></script>
      
    
  </body>
</html>